model-server/src/main/kotlin/org/modelix/model/server/store/IgniteStoreClient.kt

```diff
@@ -14,29 +14,35 @@
  */
 package org.modelix.model.server.store
 
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.withContext
 import mu.KotlinLogging
 import org.apache.ignite.Ignite
 import org.apache.ignite.IgniteCache
 import org.apache.ignite.Ignition
 import org.modelix.kotlin.utils.ContextValue
-import org.modelix.model.IKeyListener
+import org.modelix.model.IGenericKeyListener
 import org.modelix.model.persistent.HashUtil
+import org.modelix.model.server.SqlUtils
 import java.io.File
 import java.io.FileReader
 import java.io.IOException
 import java.util.*
-import java.util.stream.Collectors
+import javax.sql.DataSource
 
 private val LOG = KotlinLogging.logger { }
 
-class IgniteStoreClient(jdbcConfFile: File? = null, inmemory: Boolean = false) : IStoreClient, AutoCloseable {
+class IgniteStoreClient(jdbcConfFile: File? = null, inmemory: Boolean = false) : IGenericStoreClient<ObjectInRepository>, AutoCloseable {
 
     companion object {
         private const val ENTRY_CHANGED_TOPIC = "entryChanged"
     }
 
     private lateinit var ignite: Ignite
-    private val cache: IgniteCache<String, String?>
+    private val legacyCache: IgniteCache<String, String?>
+    private val mutableObjectsCache: IgniteCache<ObjectInRepository, String?>
```

Comment:
idea: If we copy over all keys from `legacyCache` as global keys into `mutableObjectsCache` at startup (aka. migration phase), then we could get rid of `legacyCache`.

This is already done by only writing the newly created gloabal immutable keys only to the `mutableObjectsCache`.

See `igniteStoreClient.putAll`

I think simpler code and not having the keys scattared around multiple caches would be easier to understand and debug.

Reply:
I reverted back to a single cache. All data remains in the single existing table. No copying necessary.